### PR TITLE
runtime: use class_getName instead of NSStringFromClass in _swift_checkClassAndWarnForKeyedArchiving

### PR DESCRIPTION
**Pull request by eeckstein**

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci Please smoke test and merge

File: stdlib/public/SDK/Foundation/CheckClass.mm [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 
+#include <objc/runtime.h>
+
 #include "swift/Runtime/Metadata.h"
 
+// Aliases for Objective-C runtime entry points.
+static const char *class_getName(const swift::ClassMetadata* type) {
+  return class_getName(
+    reinterpret_cast<Class>(const_cast<swift::ClassMetadata*>(type)));
+}
```
Comment (jrose-apple) [inline stdlib/public/SDK/Foundation/CheckClass.mm]: You don't need this part; you already have the metadata as an Objective-C `Class`.

Comment (eeckstein) [inline stdlib/public/SDK/Foundation/CheckClass.mm]: Yeah, I should read the code and not just copy-paste

Comment (eeckstein) [inline stdlib/public/SDK/Foundation/CheckClass.mm]: https://github.com/apple/swift/pull/10230

**Pull request merged**